os/hal/ports/STM32/LLD/FDCANv1/hal_can_lld.c

```diff
@@ -276,15 +276,22 @@ bool can_lld_start(CANDriver *canp) {
   }
 
   /* Configuration can be performed now.*/
-  canp->fdcan->CCCR   = FDCAN_CCCR_CCE;
+  canp->fdcan->CCCR  |= FDCAN_CCCR_CCE;
+  canp->fdcan->CCCR &= ~(FDCAN_CCCR_CSR | FDCAN_CCCR_CSA);
 
   /* Setting up operation mode except driver-controlled bits.*/
   canp->fdcan->NBTP   = canp->config->NBTP;
   canp->fdcan->DBTP   = canp->config->DBTP;
   canp->fdcan->CCCR  |= canp->config->CCCR & ~(FDCAN_CCCR_CSR | FDCAN_CCCR_CSA |
                                                FDCAN_CCCR_CCE | FDCAN_CCCR_INIT);
   canp->fdcan->TEST   = canp->config->TEST;
+#ifdef STM32G4XX
   canp->fdcan->RXGFC  = canp->config->RXGFC;
+#elif defined(STM32H7XX)
+  canp->fdcan->GFC    = canp->config->RXGFC;

```

Comment:
this register changed name for no apparent reason?

Reply:
I'm also not sure if this is the ChibiOS standard way to do something like this - might make more sense to put it in the registry instead.